pyiron_base/job/jobtype.py

```diff
@@ -95,8 +98,50 @@ def __new__(
             job.set_input_to_read_only()
         return job
 
+    @classmethod
+    def un_register_job_type(cls, job_name):
```

Comment:
```suggestion
    def unregister(cls, job_name_or_class):
      if insinstance(job_name_or_class, type):
        job_name_or_class = job_name_or_class.__name__
```

Reply:
```python
@JobType.unregister
class Foo:
  pass
```
sounds a little smoother.

Reply:
That way, unregister has to return the class... sound a bit odd to me?

Reply:
Actually I'd prefer that in case you unregister by name, since that would allow you to check that the right one was removed once we do allow custom names.